oauth2_client/client.py

```diff
@@ -43,63 +101,68 @@ def request(self, method, url, *args, **kwargs):  # pylint: disable=arguments-di
 
         Returns:
             Response object.
+
+        Raises:
+            TokenExpiredError: 1) when token expiry detected and `expired_callback` not specified;
+                2) when 403 Forbidden status code received and `expired_callback` not specified;
+                3) when `expired_callback` is specified and more than one token re-fetch happens
+                within `REAUTH_LIMIT_SECONDS` period.
         """
         absolute_url = urljoin(self.service_host, url)
-        return super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
-
-
-def get_client(service_name):
+        try:
+            # super().request may raise `TokenExpiredError`
+            resp = super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+            if resp.status_code == 403:
+                raise TokenExpiredError(description="403 Forbidden status code response received")
+            else:
+                return resp
+        except TokenExpiredError:
+            log.debug("Access Token for app=%s expired.", self.app.name)
+            if not self.expired_callback:
```

Comment:
Please point cases where we instantiate Clients with a not-none expired callback?

Reply:
Discussed in slack, it looks like we do not need for callback to be optional.